### PR TITLE
Chang UINT64 amounts to base 10 display

### PR DESCRIPTION
**Pull request by shawnxie999**
### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

File: src/libxrpl/protocol/STInteger.cpp [hunk 1]
```diff
@@ -194,11 +194,25 @@ STUInt64::getText() const
...
+        std::string str(
+            base == 10 ? 20 : 16, 0);  // Allocate space depending on base
+        auto ret =
+            std::to_chars(str.data(), str.data() + str.size(), value, base);
+        assert(ret.ec == std::errc());
+        str.resize(std::distance(str.data(), ret.ptr));
+        return str;
+    };
+
+    if (auto const& fName = getFName(); fName == sfMaximumAmount ||
```
Comment (gregtatcam) [inline src/libxrpl/protocol/STInteger.cpp:208]: Doesn't feel right having ST* types having knowledge of specific fields. Is it the only way?

File: src/libxrpl/protocol/STParsedJSON.cpp [hunk 2]
```diff
@@ -398,8 +398,16 @@ parseLeaf(
 
                     std::uint64_t val;
 
+                    bool const useBase10 = field == sfMaximumAmount ||
```
Comment (gregtatcam) [inline src/libxrpl/protocol/STParsedJSON.cpp:401]: Same - generic type has knowledge of specific fields.

File: src/test/jtx/mpt.h [hunk 3]
```diff
@@ -97,7 +97,7 @@ struct MPTConstr
 
 struct MPTCreate
 {
-    std::optional<std::int64_t> maxAmt = std::nullopt;
+    std::optional<std::string> maxAmt = std::nullopt;
```
Comment (gregtatcam) [inline src/test/jtx/mpt.h:100]: Do we need to change the type? Can just change when the value is formatted to the field.

File: src/libxrpl/protocol/STInteger.cpp (hunk 1 above)
Comment (shawnxie999) [reply src/libxrpl/protocol/STInteger.cpp:208]: STUInt16 has something similar, so i think this is fine
```
Json::Value STUInt16::getJson(JsonOptions) const
{
    if (getFName() == sfLedgerEntryType)
    {
        auto item = LedgerFormats::getInstance().findByType(
            safe_cast<LedgerEntryType>(value_));
        if (item != nullptr)
            return item->getName();
    }
```

File: src/libxrpl/protocol/STParsedJSON.cpp (hunk 2 above)
Comment (shawnxie999) [reply src/libxrpl/protocol/STParsedJSON.cpp:401]: ditto

**Pull request merged**